Remove duplicate namespace constants from core/v1

adrianludwin opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Two namespace-related constants are duplicated in `core/v1` and `meta/v1` (`NamespaceDefault` and `NamespaceAll`). The versions from `core` were referenced about a dozen or so times before this change, while the versions from `meta` are referenced hundreds of times. This change removes the redundant constants in `core` and changes all references to `meta`.

**Special notes for your reviewer**:
Please note that this is my first contribution to K8s and I apologize for any steps I've missed.

There are a lot of broken unit tests at head; I _believe_ there are no remaining failures caused by this change but it's hard to be certain. The following integ tests pass: apimachinery, apiserver, client; the others are still running (when they complete successfully I'll update this description).

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

adrianludwin committed:
Remove duplicate namespace constants from core/v1

adrianludwin commented:
CLA: trying again

danwinship commented:
/ok-to-test
but I have no idea if the idea is right or not.
(also, what's up with the change to `pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go` ? It looks maybe regenerated-with-an-older-version-of-tools or something?)

adrianludwin commented:
> /ok-to-test
> but I have no idea if the idea is right or not.
> (also, what's up with the change to `pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go` ? It looks maybe regenerated-with-an-older-version-of-tools or something?)

Thanks Dan, I'll check out the generated file. Do you know whom to ask about whether this is a reasonable change to make?

adrianludwin commented:
> (also, what's up with the change to `pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go` ? It looks maybe regenerated-with-an-older-version-of-tools or something?)

Fixed. Not sure what happened, but I switched between building with my locally installed tools and with Docker; when I deleted the file and re-built with Docker, the diffs went away.

caesarxuchao commented:
Note that this also breaks non-k8s-family repos that depend on k8s.io/api. I don't think it's worthwhile.

adrianludwin commented:
> Note that this also breaks non-k8s-family repos that depend on k8s.io/api. I don't think it's worthwhile.

Thanks Chao. Would you be ok with a simple doc comment suggesting that people use the meta constants instead?

adrianludwin commented:
Withdrawn
